### PR TITLE
plugins: add sharding support for bugzilla

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds config sharding support for the bugzilla plugin.

/cc @stevekuznetsov @alvaroaleman

### CODE REVIEW

**Review by petr-muller — APPROVED** [review]

File: prow/plugins/config.go [hunk 1]
```diff
@@ -1806,6 +1810,53 @@ func (p *Plugins) mergeFrom(other *Plugins) error {
...
+				errs = append(errs, fmt.Errorf("found duplicate repository config for bugzilla.%s/%s", org, repo))
+				continue
+			}
+			(*p).Orgs[org].Repos[repo] = repoConfig
+		}
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
 func (c *Configuration) HasConfigFor() (global bool, orgs sets.String, repos sets.String) {
```
Comment (alvaroaleman) [inline prow/plugins/config.go:1860]: Are you going to extend `HasConfigFor` in a different PR?

File: prow/plugins/config.go [hunk 2]
```diff
@@ -1806,6 +1810,53 @@ func (p *Plugins) mergeFrom(other *Plugins) error {
...
+		if (*p).Default != nil {
+			errs = append(errs, errors.New("configuration of global default defined in multiple places"))
+		} else {
+			(*p).Default = other.Default
+		}
+	}
+	if len(other.Orgs) != 0 && (*p).Orgs == nil {
+		newConfig := (*p)
+		newConfig.Orgs = make(map[string]BugzillaOrgOptions)
+		(*p) = newConfig
```
Comment (alvaroaleman) [inline prow/plugins/config.go:1829]: Why not simply
```suggestion
		p.Orgs = make(map[string]BugzillaOrgOptions)
```

File: prow/plugins/config.go [hunk 3]
```diff
@@ -1806,6 +1810,53 @@ func (p *Plugins) mergeFrom(other *Plugins) error {
...
+	}
+	for org, orgConfig := range other.Orgs {
+		if _, ok := (*p).Orgs[org]; !ok {
+			(*p).Orgs[org] = BugzillaOrgOptions{}
+		}
```
Comment (alvaroaleman) [inline prow/plugins/config.go:1834]: Golang automatically dererences points for you if you access their properties, can we just use that (here and everywhere else)? I had to read this a couple of times to be sure you didn't want to do something special 
```suggestion
		if _, ok := p.Orgs[org]; !ok {
			p.Orgs[org] = BugzillaOrgOptions{}
		}
```

File: prow/plugins/config.go [hunk 4]
```diff
@@ -1782,6 +1782,10 @@ func (c *Configuration) mergeFrom(other *Configuration) error {
 		errs = append(errs, fmt.Errorf("failed to merge .plugins from supplemental config: %w", err))
 	}
 
+	if err := (&(c.Bugzilla)).mergeFrom(&other.Bugzilla); err != nil {
```
Comment (alvaroaleman) [inline prow/plugins/config.go:1785]:
```suggestion
	if err := c.Bugzilla.mergeFrom(&other.Bugzilla); err != nil {
```

**Comment by AlexNPavel** [comment]
Updated PR to remove weird pointer stuff I was doing and update the `HasConfigFor` function

**Review by alvaroaleman — APPROVED** [review]
/label tide/merge-method-squash

**Pull request merged**